src/Illuminate/Database/Concerns/TransactionException.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Illuminate\Database\Concerns;
+
+use Exception;
+
+class TransactionException extends Exception
+{
+}
```

Comment:
A bit odd to have `Exception` under `Concern` namespace, technically this should be under contracts or exceptions folder.

Reply:
If i understand it correctly `Contracts` is meant for interfaces only. The `Exception`-namespace was my idea as well, but i did not know where to put it. Seemed still odd in the `Concerns`-namespace.
In the end i used the `Eloquent`-namespace as inspiration and just dropped it where i used it.
Is there any special guideline for exceptions?

Reply:
https://github.com/illuminate/contracts/search?utf8=%E2%9C%93&q=Exception&type=

There are exceptions in the contracts repository

Reply:
Well...yeah. But they are used there, right? So thats the impression i got, that exceptions just are created and dropped where they are used first.
Is there a preferred way in the laravel framework to handle such things?